FreeBSD Metricbeat failure getting system process FDs

This is probably an issue with elastic/gosigar.

http://build-eu-00.elastic.co/view/Beats/job/beats-freebsd/542/console
```
--- FAIL: TestGetProcess (0.00s)
	assertions.go:225: 
                        
	Error Trace:	helper_test.go:36
		
	Error:		Received unexpected error "error getting process file descriptor usage for pid=1458: open /proc/1458/rlimit: no such file or directory"
```

--- a/glide.yaml
+++ b/glide.yaml
@@ -25,7 +25,7 @@ import:
   subpackages:
   - /difflib
 - package: github.com/elastic/gosigar
-  version: a6e5709f6a491f83588305c15151412c2e480fc2
+  version: 2716c1fe855ee5c88eae707195e0688374458c92
 - package: github.com/samuel/go-parser
   version: ca8abbf65d0e61dedf061f98bd3850f250e27539
 - package: github.com/samuel/go-thrift

--- a/metricbeat/docs/faq.asciidoc
+++ b/metricbeat/docs/faq.asciidoc
@@ -10,10 +10,12 @@ https://discuss.elastic.co/c/beats/metricbeat[Metricbeat discussion forum].
 
 The system metricsets rely a Linux comparability layer to retrieve metrics on
 FreeBSD. You need to mount the Linux procfs filesystem using the following
-commands.
+commands. You may want to add these filesystems to your `/etc/fstab` so they are
+mounted automatically.
 
 [source,sh]
 ----
+sudo mount -t procfs proc /proc
 sudo mkdir -p /compat/linux/proc
 sudo mount -t linprocfs /dev/null /compat/linux/proc
 ----

--- a/metricbeat/module/system/process/helper.go
+++ b/metricbeat/module/system/process/helper.go
@@ -6,6 +6,7 @@ import (
 	"fmt"
 	"os"
 	"regexp"
+	"runtime"
 	"strings"
 	"time"
 
@@ -99,6 +100,12 @@ func (proc *Process) getDetails(cmdline string) error {
 // data then  nil is returned with no error (/proc/[pid]/fd requires root
 // permissions). Any other errors that occur are returned.
 func getProcFDUsage(pid int) (*sigar.ProcFDUsage, error) {
+	// It's not possible to collect FD usage from other processes on FreeBSD
+	// due to linprocfs not exposing the information.
+	if runtime.GOOS == "freebsd" && pid != os.Getpid() {
+		return nil, nil
+	}
+
 	fd := sigar.ProcFDUsage{}
 	if err := fd.Get(pid); err != nil {
 		switch {

